src/Microsoft.AspNetCore.Mvc.Razor.Extensions.Version1_X/RazorExtensions.cs

```diff
@@ -37,6 +37,7 @@ public static void RegisterViewComponentTagHelpers(IRazorEngineBuilder builder)
         {
             EnsureDesignTime(builder);
 
+            builder.Features.Add(new ViewComponentTagHelperDescriptorProvider());
```

Comment:
We have a kind of a wierd thing going on where these features don't get registered exception manually. This was hardcoded in a bad way. There's no hard in us registering these features even if they go unused.